src/amplitude/event.py

```diff
@@ -587,31 +979,103 @@ def __init__(self, price: float,
         self.revenue: Optional[float] = revenue
 
     def set_receipt(self, receipt: str, receipt_signature: str):
+        """Set the receipt and signature
+
+        Args:
+            receipt (str): The receipt identifier
+            receipt_signature (str): The receipt signature
+
+        Returns:
+            the Revenue instance itself
+        """
         self.receipt = receipt
         self.receipt_sig = receipt_signature
         return self
 
     def is_valid(self):
+        """Check if a revenue instance is a valid one
+
+        Returns:
+            True if price is a float number, False otherwise
+        """
         return isinstance(self.price, float)
```

Comment:
Should we check quantity here as well?

Reply:
yes, I added check for int type and positive